### PR TITLE
Improve update device location 

### PR DESCRIPTION
**Pull request by CameronRP**
- Send more info with location (altitude, accuracy, time)
- Don't send location if lat and long are 0
- Update DB version to fix app from crashing when running on a phone that ran a version with the old DB

### CODE REVIEW

File: app/src/main/java/nz/org/cacophony/sidekick/MainActivity.kt [hunk 1]
```diff
@@ -299,31 +302,81 @@ class MainActivity : AppCompatActivity() {
...
 
         }
-        locationRequest = LocationRequest.create()?.apply {
+        createLocationRequest()
+    }
+
+    private fun createLocationRequest() {
+        val locationRequest = LocationRequest.create().apply {
+            interval = 10000
+            fastestInterval = 5000
```
Comment (mjs) [inline app/src/main/java/nz/org/cacophony/sidekick/MainActivity.kt]: good choices

File: app/src/main/java/nz/org/cacophony/sidekick/MainActivity.kt [hunk 2]
```diff
@@ -299,31 +302,81 @@ class MainActivity : AppCompatActivity() {
...
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        when(requestCode) {
+            locationSettingsUpdateCode -> {
+                if (resultCode == -1) {
+                    createLocationRequest()
+                } else {
+                    makeToast( "Don't have proper location settings to get location.")
```
Comment (mjs) [inline app/src/main/java/nz/org/cacophony/sidekick/MainActivity.kt]: Is this a permissions thing or more general? If it's about permission you should probably use the word "permissions" here.

Comment (CameronRP) [reply app/src/main/java/nz/org/cacophony/sidekick/MainActivity.kt]: This is about the Google's location service that is a phones setting not an apps permission.

Comment (mjs) [reply app/src/main/java/nz/org/cacophony/sidekick/MainActivity.kt]: ok, that's fine then.

**Pull request merged**